docs/modules/plugins/pages/plugin-json.adoc

```diff
@@ -182,3 +182,138 @@ When I issue a HTTP GET request for a resource with the URL 'http://jsonpath.her
 When I convert JSON from context to scenario variable `jsonData`
 Then `${jsonData.store.book[1].price}` is = `12.99`
 ----
+
+=== Perform steps on elements in JSON
+
+Performs steps against all elements found by JSON path in JSON data or the context.
+
+Actions performed by step:
+
+* searches for elements using JSON path
+* checks that elements quantity matches comparison rule and elements number
+* passes if the comparison rule matches and the elements number is 0
+* for each element switches JSON context and performs all steps. No steps will be performed in case of comparison rule mismatch
+* restores previously set context
+
+.JSON based step
+[source,gherkin]
+----
+When I find $comparisonRule `$elementsNumber` JSON elements from `$json` by `$jsonPath` and for each element do$stepsToExecute
+----
+
+* `comparisonRule` - xref:parameters:comparison-rule.adoc[comparison rule]
+* `elementsNumber` - expected number of elements
+* `json` - JSON data
+* `jsonPath` - {json-path}
+* `stepsToExecute` - steps to perform on JSON elements
+
+.Context based step
+[source,gherkin]
+----
+When I find $comparisonRule `$elementsNumber` JSON elements from context by `$jsonPath` and for each element do$stepsToExecute
+----
+
+* `comparisonRule` - xref:parameters:comparison-rule.adoc[comparison rule]
+* `elementsNumber` - expected number of elements
+* `json` - JSON data
+* `jsonPath` - {json-path}
+* `stepsToExecute` - steps to perform on JSON elements
+
+.Verify each account id is a number
+[source,gherkin]
+----
+When I find > `0` JSON elements from `
+{
+  "accounts": [
+    {
+      "accountId": 00,
+      "status": "Active"
+    },
+    {
+      "accountId": 01,
+      "status": "Active"
+    },
+    {
+      "accountId": 10,
+      "status": "Active"
+    }
+  ]
+}
+` by `$.accounts.*` and for each element do
+|step                                                                                 |
+|Then number of JSON elements by JSON path `$[?(@.accountId =~ /\d+/i)]` is equal to 1|
+When I send HTTP GET to the relative URL 'api/v3/accounts'
+When I find > `0` JSON elements from context by `$.accounts.*` and for each element do
+|step                                                                                 |
+|Then number of JSON elements by JSON path `$[?(@.accountId =~ /\d+/i)]` is equal to 1|
```

Comment:
should no we separate these two examples on finding by JSON path in JSON data or the context.